opm/autodiff/createGlobalCellArray.hpp

```diff
@@ -34,7 +34,7 @@ void createGlobalCellArray(const Grid &grid, std::vector<int>& dest)
     dest.resize(numCells);
     const auto& globalCell = Opm::UgGridHelpers::globalCell(grid);
     std::vector<int> compressedToCartesianIdx(numCells);
-    for (unsigned cellIdx = 0; cellIdx < numCells; ++cellIdx) {
+    for (int cellIdx = 0; cellIdx < numCells; ++cellIdx) {
```

Comment:
May I suggest a somewhat more substantial rewrite of this function?  Checking a constant condition (`if (globalCell)`) on every iteration just looks wrong.  Using the `compressedToCartesian()` function from OPM/opm-core#879, possibly renamed, is the right thing to do here.